File: Summer/app/src/main/java/com/prim/honorkings/MainActivity.java
package com.prim.honorkings;

import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;
import android.os.Message;

import androidx.annotation.Nullable;

import com.prim.honorkings.logic.MainActivityLogic;
import com.prim.summer_common.ui.component.SummerBaseActivity;

public class MainActivity extends SummerBaseActivity implements MainActivityLogic.ActivityProvider {

    private MainActivityLogic activityLogic;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_tab_bottom);
        activityLogic = new MainActivityLogic(this,savedInstanceState);
        Handler handler = new Handler(Looper.myLooper(), new Handler.Callback() {
            @Override
            public boolean handleMessage(Message msg) {


                return false;
            }
        });
    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
        activityLogic.onSaveInstanceState(outState);
    }
}

File: Summer/app/src/main/java/com/prim/honorkings/MainActivity.kt
package com.prim.honorkings

import android.os.Bundle
import com.prim.honorkings.logic.MainActivityLogic
import com.prim.honorkings.logic.MainActivityLogic.ActivityProvider
import com.prim.summer_common.ui.component.SummerBaseActivity

class MainActivity : SummerBaseActivity(), ActivityProvider {
    private var activityLogic: MainActivityLogic? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_tab_bottom)
        activityLogic = MainActivityLogic(this, savedInstanceState)
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        activityLogic?.onSaveInstanceState(outState)
    }
}